Description: Correct lsb runlevels and dependencies in initscript
Author: Petter Reinholdtsen <pere@hungry.com>
Bug-Debian: https://bugs.debian.org/542283

Index: flamethrower-0.1.8/etc/init.d/flamethrower-server
===================================================================
--- flamethrower-0.1.8.orig/etc/init.d/flamethrower-server
+++ flamethrower-0.1.8/etc/init.d/flamethrower-server
@@ -16,13 +16,12 @@
 # Support for LSB compliant init system:
 ### BEGIN INIT INFO
 # Provides: flamethrowerd
-# Required-Start: $network $syslog
-# Required-Stop:
-# Default-Start:  3 5
-# Default-Stop:   0 1 2 6
+# Required-Start: $remote_fs $syslog $network
+# Required-Stop:  $remote_fs $syslog
+# Default-Start:  2 3 4 5
+# Default-Stop:   0 1 6
 # Short-Description: Daemon for multicast file distribution
 # Description: Daemon for multicast file distribution
-#
 ### END INIT INFO
 
 kill_em_all() {
