Description: Fix compatibility with MakeMaker changes in perl 5.22
Author: gregor herrmann <gregoa@debian.org>
Bug-Debian: http://bugs.debian.org/792382
Forwarded: no

Index: flamethrower-0.1.8/Makefile.PL
===================================================================
--- flamethrower-0.1.8.orig/Makefile.PL
+++ flamethrower-0.1.8/Makefile.PL
@@ -9,9 +9,9 @@ sub MY::postamble {
     my $string = <<EOF;
 
 set_ver::
-\t${install} \$(PREFIX)/bin/flamethrower
-\t${install} \$(PREFIX)/bin/flamethrowerd
-\t${install} \$(PREFIX)/lib/flamethrower/Flamethrower.pm
+\t${install} \$(DESTDIR)/\$(PREFIX)/bin/flamethrower
+\t${install} \$(DESTDIR)/\$(PREFIX)/bin/flamethrowerd
+\t${install} \$(DESTDIR)/\$(PREFIX)/lib/flamethrower/Flamethrower.pm
 
 install_configs::
 \tinstall -d \$(CONFDIR)/etc/init.d/
