Description: Fix typos
Author: dann frazier <dannf@debian.org>
Forwarded: Fixed in upstream repo before it vanished

Index: flamethrower-0.1.8/README
===================================================================
--- flamethrower-0.1.8.orig/README
+++ flamethrower-0.1.8/README
@@ -3,7 +3,7 @@
 #
 #   Copyright (C) 2003 Brian Elliott Finley <finley@mcs.anl.gov>
 #
-#   $Id: README 32 2003-07-01 03:54:42Z dannf $
+#   $Id: README 60 2006-04-18 23:07:43Z dannf $
 #
 
 What it is:
@@ -16,7 +16,7 @@ What it is:
 
 Notable characteristics:
 
-    1)  Works with entire directory heirarchies of files, not just single files.
+    1)  Works with entire directory hierarchies of files, not just single files.
     2)  Uses a server configuration file that takes module entries that are 
         similar to those used by rsyncd.conf.
 
Index: flamethrower-0.1.8/bin/flamethrowerd
===================================================================
--- flamethrower-0.1.8.orig/bin/flamethrowerd
+++ flamethrower-0.1.8/bin/flamethrowerd
@@ -5,7 +5,7 @@
 #   Copyright (C) 2003 Bald Guy Software 
 #                      Brian Elliott Finley <brian@bgsw.net>
 #
-#   $Id: flamethrowerd 45 2003-11-17 00:12:55Z brianfinley $
+#   $Id: flamethrowerd 59 2006-04-18 23:05:44Z dannf $
 #
 
 $0 = "flamethrowerd";
@@ -343,7 +343,7 @@ sub update_directory {
     # Create our directory directory, if necessary.
     my $dir = $ft_config->get("flamethrower_directory_dir");
     if ( ! -e $dir ) {
-        mkdir("$dir", 0755) or die("Couddn't mkdir $dir!");
+        mkdir("$dir", 0755) or die("Couldn't mkdir $dir!");
     }
     unlink <$dir/*>;
 
